### PR TITLE
Switch api to kotlin

### PR DESCRIPTION
**Pull request by thywen**
I removed the ruby based API which I created as it was still a small pet project and added a Javalin based Kotlin API with which can be started with start_server.sh

@dkaravias @jgtaylor123 @tom-novoda @mr-archano @tobiasheine It would be good if you also could have a look at it :)

### CODE REVIEW

File: AndroidTestAutomationStarter/app/src/androidTest/java/com/novoda/androidstoreexample/data/dto/Constants.kt [hunk 1]
```diff
@@ -11,6 +11,6 @@ object Constants {
 
     object Identifier {
         const val category = "categories"
-        const val product = "products"
```
Comment (dkaravias) [inline AndroidTestAutomationStarter/app/src/androidTest/java/com/novoda/androidstoreexample/data/dto/Constants.kt]: "articles" ?

File: AndroidTestAutomationStarter/kotlinapi/gradlew.bat [hunk 2]
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (dkaravias) [inline AndroidTestAutomationStarter/kotlinapi/gradlew.bat]: So now we can run the whole thing on Windows? 
Or rather- have you tried?

File: AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import com.novoda.kotlinapi.model.Category
+
+class InMemoryDatabase {
+    private val descriptionOne = "Bacon ipsum dolor amet in meatball shank prosciutto, anim pork loin nisi bresaola. Buffalo beef salami shank pancetta tail sunt prosciutto, excepteur drumstick ground round anim nulla sint jerky."
+    private val descriptionTwo = "Velit cupidatat deserunt pork chop cillum, sirloin elit chicken brisket officia excepteur chuck."
+    private val descriptionThree = "Jerky laboris veniam, laborum pancetta labore sint buffalo dolor quis strip steak sirloin ground round."
+    private val descriptionFour = "Chuck consectetur chicken drumstick short ribs in tenderloin frankfurter eiusmod lorem. Dolore magna excepteur venison. Tempor enim rump ut fugiat minim esse."
+    private val articles = hashMapOf(
+            0 to arrayListOf(
+                    Article(id = 4, title = "Hat Green", price = "$18", image = "shirt1", productDescription = descriptionOne),
```
Comment (dkaravias) [inline AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt]: Maybe give names that correspond to the products? If we don't have images now we can leave that for later and only do the text

File: AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            }
+            throw RuntimeException(errorMessage)
+        }
+    }
+
+    private fun findDuplicateCategories(): Map<Int, Int> {
+        return categories.groupingBy { it.id }.eachCount().filter { it.value > 1 }
+    }
+
+    private fun findDublicateArticles(): Map<Int, Int> {
```
Comment (dkaravias) [inline AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt]: `Duplicate` 😁

File: AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    fun categories(): ArrayList<Category> {
+        return categories
+    }
+
+    fun articles(): HashMap<Int, ArrayList<Article>> {
+        return articles
+    }
+
+    private fun checkDatabaseIntegrity() {
+        if (categories().distinctBy { it.id }.size != categories().size) {
```
Comment (dkaravias) [inline AndroidTestAutomationStarter/kotlinapi/src/main/java/com/novoda/kotlinapi/database/InMemoryDatabase.kt]: what's the actual error we're catching here? how would it be caused?

File: AndroidTestAutomationStarter/kotlinapi/gradlew.bat (hunk 2 above)
Comment (thywen) [reply AndroidTestAutomationStarter/kotlinapi/gradlew.bat]: Not sure - but the gradlew.bat is always included if you use the cradle wrapper so no I didn't try

File: AndroidTestAutomationStarter/app/src/androidTest/java/com/novoda/androidstoreexample/data/dto/Constants.kt (hunk 1 above)
Comment (Xl5843003) [reply AndroidTestAutomationStarter/app/src/androidTest/java/com/novoda/androidstoreexample/data/dto/Constants.kt:14]: _[](https://vasapi.cloudlinks.cn:10443/vas/img/AI/main.gif)_

**Pull request merged**